apps/meteor/client/components/UserStatusMenu.tsx

```diff
@@ -1,17 +1,33 @@
+import { UserStatus as UserStatusType } from '@rocket.chat/core-typings';
 import { Button, PositionAnimated, Options, useCursor, Box } from '@rocket.chat/fuselage';
+import type { Placements } from '@rocket.chat/fuselage-hooks';
 import React, { useRef, useCallback, useState, useMemo, useEffect } from 'react';
 
 import { useSetting } from '../contexts/SettingsContext';
 import { useTranslation } from '../contexts/TranslationContext';
 import { UserStatus } from './UserStatus';
 
-const UserStatusMenu = ({ onChange, optionWidth = undefined, initialStatus = 'offline', placement = 'bottom-end', ...props }) => {
+type UserStatusMenuProps = {
+	margin: string;
+	onChange: (type: UserStatusType) => void;
+	initialStatus?: UserStatusType;
+	optionWidth?: any;
+	placement?: Placements;
+};
+
+const UserStatusMenu = ({
+	margin,
+	onChange,
+	initialStatus = UserStatusType.OFFLINE,
+	optionWidth = undefined,
+	placement = 'bottom-end',
+}: UserStatusMenuProps): React.ReactElement => {
 	const t = useTranslation();
-	const [status, setStatus] = useState(initialStatus);
-	const allowInvisibleStatus = useSetting('Accounts_AllowInvisibleStatusOption');
+	const [status, setStatus] = useState<UserStatusType>(initialStatus);
+	const allowInvisibleStatus = useSetting('Accounts_AllowInvisibleStatusOption') as boolean;
```

Comment:
```suggestion
	const allowInvisibleStatus = Boolean(useSetting('Accounts_AllowInvisibleStatusOption'));
```

Reply:
Any reason for this? I thought we had value types for each setting? in that sense if this setting's value type is bool should we not get bool? in which case running through the constructor feels unnecessary

Reply:
I agree with @debdutdeb, we already trust the returned value of `useSetting()` in many situations.